qutip/solver/options.py

```diff
@@ -0,0 +1,258 @@
+__all__ = ['SolverOptions',
+           'SolverResultsOptions', 'SolverOdeOptions',
+           'McOptions']
+
+from ..optionsclass import optionsclass
+import multiprocessing
+
+@optionsclass("solver")
+class SolverOptions:
+    """
+    Class of options for evolution solvers such as :func:`qutip.mesolve` and
+    :func:`qutip.mcsolve`. Options can be specified either as arguments to the
+    constructor::
+
+        opts = SolverOptions(progress_bar='enhanced', ...)
+
+    or by changing the class attributes after creation::
+
+        opts = SolverOptions()
+        opts['progress_bar'] = 'enhanced'
+
+    Returns options class to be used as options in evolution solvers.
+
+    The default can be changed by::
+
+        qutip.settings.solver['progress_bar'] = 'enhanced'
+
+    Options
+    -------
+    progress_bar : str {'text', 'enhanced', 'tqdm', ''}
+        How to present the solver progress.
+        'tqdm' use the python module of the same name and raise an error if
+        not installed.
+        True will result in 'text'.
+        Empty string or False will disable the bar.
+
+    progress_kwargs : dict
+        kwargs to pass to the progress_bar. Qutip's bars use `chunk_size`.
+    """
+    options = {
+        # (turned off for batch unitary propagator mode)
+        "progress_bar": "text",
+        # Normalize output of solvers
+        # (turned off for batch unitary propagator mode)
+        "progress_kwargs": {"chunk_size":10},
+    }
+
+
+@optionsclass("ode", SolverOptions)
+class SolverOdeOptions:
+    """
+    Class of options for evolution solvers such as :func:`qutip.mesolve` and
+    :func:`qutip.mcsolve`. Options can be specified either as arguments to the
+    SolverOptions constructor::
+
+        opts = SolverOptions(method=bdf, ...)
+
+    or by changing the class attributes after creation::
+
+        opts = SolverOptions()
+        opts.ode['method'] = 'bdf'
+
+    Returns options class to be used as options in evolution solvers.
```

Comment:
The start of the docstring seems to be a copy and paste of the text from the `SolverOptions` docstring. This one should say something different. :)